src/devices/src/virtio/block/request.rs

```diff
@@ -178,52 +211,85 @@ impl Request {
         Ok(req)
     }
 
+    fn execute_seek(&self, disk: &mut DiskProperties) -> result::Result<(), ErrStatus> {
+        // Check that the data length is a multiple of 512 as specified in the virtio standard.
+        // We could also check this at request parsing level.
+        if u64::from(self.data_len) & (SECTOR_SIZE - 1) != 0 {
```

Comment:
nit: I believe the compiler optimizes the `%` operation anyway, so why not use that to make it more clear?

Reply:
Yes, it doesn't seem to affect the performance. Done.